### ISSUE TITLE
Support Vault for encryption

### ISSUE THREAD

**Issue by sixfeetover** [mozilla/sops#291]
Would love to see support for [Vault's transit backend](https://www.vaultproject.io/docs/secrets/transit/index.html) as the encryption key authority!

**Comment by benlangfeld** [comment]
Are any of the core team able to provide some pointers on how this should be implemented. My team might be able to take a stab at this if we can get some help.

**Comment by autrilla** [comment]
Sure! Depending on what you want, you either have to implement [the Cipher interface](https://github.com/mozilla/sops/blob/master/sops.go#L79) (see [example](https://github.com/mozilla/sops/blob/master/aes/cipher.go)) or [the MasterKey interface](https://github.com/mozilla/sops/blob/master/keys/keys.go#L4) (see [example](https://github.com/mozilla/sops/pull/245)). The former would make Vault an alternative for AES, whereas the latter would make it an alternative for GPG/KMS. Is this enough to get you started?

**Comment by Just-Insane** [comment]
Has there been any further work on this issue?

**Comment by jvehent** [comment]
The core team isn't working on this feature but we'd happily review and merge a patch from a motivated contributor.

**Comment by binlab** [comment]
Good idea! Any progress with it?

**Comment by Just-Insane** [comment]
No, I built [Helm-Vault](https://github.com/Just-Insane/Helm-Vault) instead.

**Comment by binlab** [comment]
As I see this implementation almost done - https://github.com/mozilla/sops/pull/623

**Comment by smnbbrv** [comment]
Looks like it's already implemented and merged https://github.com/mozilla/sops/pull/655. The question is when is it going to be released?

**Comment by Just-Insane** [comment]
This appears to just use Vault as a KMS storage area for your encryption keys, and does not actually store your secret values in Vault (which may be the preferred method for some).

**Comment by autrilla** [comment]
> This appears to just use Vault as a KMS storage area for your encryption keys, and does not actually store your secret values in Vault (which may be the preferred method for some).

Yes, sops publish does what you want instead

**Comment by Just-Insane** [comment]
Link to the documentation on sops publish?

**Comment by autrilla** [comment]
> Link to the documentation on sops publish?

https://github.com/mozilla/sops/blob/master/README.rst#2171publishing-to-vault
